part3/part3_demo.ipynb

```
@@ -0,0 +1,157 @@
+    "    print(\"✅ PASS: Đã giải đúng hệ chéo trội.\\n\")\n",
+    "else:\n",
+    "    print(\"❌ FAIL: Kết quả lệch.\\n\")\n",
+    "\n",
+    "# ---------------------------------------------------------\n",
+    "# Test 2: Ma trận đối xứng xác định dương (SPD - Symmetric Positive Definite)\n",
+    "# ---------------------------------------------------------\n",
+    "print(\"Test 2: SPD (Symmetric Positive Definite)\")\n",
+    "# Ma trận SPD không nhất thiết phải chéo trội nghiêm ngặt, nhưng Gauss-Seidel vẫn hội tụ.\n",
+    "A2 = [[2.0, -1.0, 0.0], \n",
+    "      [-1.0, 2.0, -1.0], \n",
+    "      [0.0, -1.0, 2.0]]\n",
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)